core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -120,19 +120,44 @@ public TenantObject setEnabled(final boolean flag) {
      * @param subjectDn The CA's subject DN.
      * @return This tenant for command chaining.
      * @throws NullPointerException if any of the parameters is {@code null}.
+     * @deprecated Use {@link #addTrustAnchor(PublicKey, X500Principal, Boolean)} instead.
```

Comment:
I do not see why this should be deprecated. Its original purpose is still valid, isn't it? We should document, though, that it clears the set of trust anchors and adds the given one but doesn't allow it to be used for auto provisioning.

Reply:
FMPOV the add operation is more useful since Hono supports multiple trust anchors. I do not see anything special for auto-provisioning. Both methods are only used in tests. But on the other hand, it does not hurt to keep it.